compiler/src/iree/compiler/Codegen/Dialect/VectorExt/IR/VectorExtAttrs.cpp

```diff
@@ -52,13 +52,13 @@ std::optional<int64_t> LayoutAttr::getShape(const LayoutDimension &dim) const {
 
 // Get the SIMT Vector shape in the order specified by dims. If no dims are
 // specified, then return an empty vector.
-LogicalResult LayoutAttr::isValidLayout(VectorValue vector) const {
-  ArrayRef<int64_t> shape = vector.getType().getShape();
+LogicalResult LayoutAttr::isValidLayout(ShapedType shapeTy,
+                                        Location loc) const {
+  ArrayRef<int64_t> shape = shapeTy.getShape();
   if (shape.size() != getRank()) {
-    return emitError(vector.getLoc(), "Rank of vector (" +
-                                          std::to_string(shape.size()) +
-                                          ") does not match rank of layout (" +
-                                          std::to_string(getRank()) + ").");
+    return emitError(loc, "Rank of vector (" + std::to_string(shape.size()) +
+                              ") does not match rank of layout (" +
```

Comment:
I think you can do `emirError(loc, "Foo") << bar << "baz";`, no? If not, we have `llvm::formatv`

Reply:
I can apparently! I didnt know this. Thanks!